scripts/gen-edpm-bmset-password-secret.sh

```diff
@@ -24,7 +24,7 @@ metadata:
   name: baremetalset-password-secret
   namespace: ${NAMESPACE}
 type: Opaque
-data:
+stringData:
```

Comment:
Ok I see everything starts with `BM_ROOT_PASSWORD` passed as input, that is propagated here.
I don't have big concerns to prefer `data` over `stringData` (as we're delegating k8s to encode the secret for us), so I'm ok with the patch if it ease the development workflow.

Reply:
@abays FYI ^

Reply:
Does our CI automation currently use this?  I just wonder if they're already passing base64-encoded values through the env var (`BM_ROOT_PASSWORD`), such that if it was injected into `stringData`, k8s would try to  base64-encode something that was already encoded by the user.

Reply:
Good question, in that case we need to update how CI pass this info as well. @rabi I'm wondering if we want to clarify this aspect before merging.

Reply:
I had checked it before proposing. I didn't find it being used in any of the `openstack-k8s-operator` repos. Unless there is some other places that we should look for.

Reply:
ack, feel free to remove the hold when you're comfortable merging the patch then.

Reply:
I really don't know if something in CI is using this already or not.  It just seems possible to me that some existing job automation might be setting `BM_ROOT_PASSWORD` as an encoded string already.  Maybe we should ask someone on the CI side to comment?  @raukadah, maybe you know something about this?

Reply:
I don't see it being used in upstream/downstream CI.